packages/esm-patient-chart-app/src/visit/visits-widget/queue.resource.tsx

```diff
@@ -0,0 +1,151 @@
+export type QueueStatus = 'Finished Service' | 'In Service' | 'Waiting';
+
+interface VisitQueueEntry {
+  queueEntry: VisitQueueEntry;
+  uuid: string;
+  visit: Visit;
+}
+
+interface VisitQueueEntry {
+  display: string;
+  endedAt: null;
+  locationWaitingFor: string | null;
+  patient: {
+    uuid: string;
+    person: {
+      age: string;
+      gender: string;
+    };
+    phoneNumber: string;
+  };
+  priority: {
+    display: QueuePriority;
+    uuid: string;
+  };
+  priorityComment: string | null;
+  providerWaitingFor: null;
+  queue: {
+    description: string;
+    display: string;
+    name: string;
+    service: {
+      display: QueueService;
+    };
+    uuid: string;
+  };
+  startedAt: string;
+  status: {
+    display: QueueStatus;
+    uuid: string;
+  };
+  uuid: string;
+  visit: Visit;
+}
+
+export interface MappedVisitQueueEntry {
+  id: string;
+  encounters: Array<MappedEncounter>;
+  name: string;
+  patientUuid: string;
+  priority: MappedQueuePriority;
+  priorityComment: string;
+  priorityUuid: string;
+  service: QueueService;
+  status: QueueStatus;
+  statusUuid: string;
+  visitStartDateTime: string;
+  visitType: string;
+  visitUuid: string;
+  waitTime: string;
+  queueUuid: string;
+  queueEntryUuid: string;
+}
+interface ObsData {
+  concept: {
+    display: string;
+    uuid: string;
+  };
+  value?: string | any;
+  groupMembers?: Array<{
+    concept: { uuid: string; display: string };
+    value?: string | any;
+  }>;
+  obsDatetime: string;
+}
+
+interface UseVisitQueueEntries {
+  visitQueueEntries: Array<MappedVisitQueueEntry> | null;
+  isLoading: boolean;
+  isError: Error;
+  isValidating?: boolean;
+}
+
+interface Encounter {
+  diagnoses: Array<any>;
+  encounterDatetime: string;
+  encounterProviders?: Array<{ provider: { person: { display: string } } }>;
+  encounterType: { display: string; uuid: string };
+  obs: Array<ObsData>;
+  uuid: string;
+  voided: boolean;
+}
+interface MappedEncounter extends Omit<Encounter, 'encounterType' | 'provider'> {
+  encounterType: string;
+  provider: string;
+}
+export function useVisitQueueEntries(): UseVisitQueueEntries {
+  const apiUrl = `/ws/rest/v1/visit-queue-entry?v=full`;
+  const { data, error, isValidating } = useSWR<{ data: { results: Array<VisitQueueEntry> } }, Error>(
+    apiUrl,
+    openmrsFetch,
+  );
+
+  const mapEncounterProperties = (encounter: Encounter): MappedEncounter => ({
+    diagnoses: encounter.diagnoses,
+    encounterDatetime: encounter.encounterDatetime,
+    encounterType: encounter.encounterType.display,
+    obs: encounter.obs,
+    provider: encounter.encounterProviders[0]?.provider?.person?.display,
+    uuid: encounter.uuid,
+    voided: encounter.voided,
+  });
+
+  const mapVisitQueueEntryProperties = (visitQueueEntry: VisitQueueEntry): MappedVisitQueueEntry => ({
+    id: visitQueueEntry.queueEntry.uuid,
+    encounters: visitQueueEntry.visit?.encounters?.map(mapEncounterProperties),
```

Comment:
perhaps you only need the status, service and priority info, the rest is not used